app/code/Magento/Catalog/Block/Product/ImageFactory.php

```diff
@@ -160,6 +160,7 @@ public function create(Product $product, string $imageId, array $attributes = nu
             );
         }
 
+        $attributes = $attributes === null ? [] : $attributes;
```

Comment:
Simply cast `$attributes` to array in place.

After changes are applied and all builds are green, please squash them into a single commit so that we have perfectly clean history 😉